tests/run-make/broken-pipe-no-ice/rmake.rs

```diff
@@ -0,0 +1,50 @@
+//! Check that `rustc` and `rustdoc` does not ICE upon encountering a broken pipe due to unhandled
+//! panics from raw std `println!` usages.
+//!
+//! Regression test for <https://github.com/rust-lang/rust/issues/34376>.
+
+// FIXME(jieyouxu): should test for Windows too, but rustc currently has a hack on Windows that
+// suppresses broken pipe errors.
+//@ only-unix (related to `-Zon-broken-pipe` behavior)
+//@ ignore-cross-compile (needs to run test binary)
+
+use std::io::Read;
+use std::os::unix::process::ExitStatusExt;
+use std::process::{Command, Stdio};
+
+use run_make_support::{env_var, libc};
+
+fn check_broken_pipe_handled_gracefully(name: &str, mut cmd: Command) {
+    cmd.stdout(Stdio::piped()).stderr(Stdio::piped());
+
+    let mut child = cmd.spawn().unwrap();
+    // Close stdout
+    drop(child.stdout.take());
```

Comment:
I would recommend caution with this approach, since it is inherently racey. For example, it assumes that it can close the handle before `rustc` is able to print. There can also be assumptions here about `PIPE_BUF` and other pipe behaviors. This can be unreliable on some systems. For example, on macOS, this should fail occasionally. 

That's why cargo's tests which exercise this same behavior use a proc-macro to synchronize with `rustc`. (see [here](https://github.com/rust-lang/cargo/blob/a06986cda652fb9d91ce027977d2126efa111e4f/tests/testsuite/build.rs#L6576-L6776)). That way the test can ensure that the handle is closed *before* `rustc` prints. Those tests also have some comments about the hazards of how this works.

Reply:
Thanks for the context, I see why that test is so complex now...

Reply:
Adjusted the test to use anonymous pipes instead which will make sure stdout is broken before the command is spawned.